src/profile-logic/import/chrome.js

```diff
@@ -678,10 +680,28 @@ async function processTracingEvents(
       }
 
       if (profileEvent.name === 'CpuProfile') {
-        profile.meta.profilingStartTime =
-          profileEvent.args.data.cpuProfile.startTime / 1000;
+        // Get profile start time from profile URL if available. Timestamp format is what
+        // is generated by the Chrome DevTools UI when saving profiles. Technically, that
+        // timestamp is of profile save instead of profile start, but one can generate a
+        // more accurate start timestamp through e.g. handling the Chrome DevTools
+        // Protocol `Profiler.consoleProfileStarted` event.
+        const timeStampRe =
+          /(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2}).*\.cpuprofile(?:$|\/|\?|#)/;
+        const match = timeStampRe.exec(profileUrl ?? '');
+        if (match) {
+          const dateTimeString = `${match[1]}-${match[2]}-${match[3]}T${match[4]}:${match[5]}:${match[6]}`;
+          const startTime = new Date(dateTimeString);
+          profile.meta.startTime = startTime.getTime();
```

Comment:
Note that `startTime` represent the moment the main thread started. So it's not great. But I don't see a better option with our current code.

Reply:
As long as the (adjustable) start time provided is reasonably close to the actual start time, I think it's going to be ok. Support for milliseconds could be added later on for more accuracy.